bitcoin/src/psbt/raw.rs

```diff
@@ -68,7 +67,10 @@ where
 
 impl fmt::Display for Key {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        write!(f, "type: {:#x}, key: {:x}", self.type_value, self.key.as_hex())
+        let mut key_bytes = vec![0u8; self.key.len() + 1];
+        key_bytes[0] = self.type_value;
+        key_bytes[1..].copy_from_slice(&self.key);
+        write!(f, "{}", key_bytes.as_hex())
```

Comment:
This will display just the hex key when we do the call-through to `to_string` from `to_hex`. Makes sense to display one after another, I'll update it to do that

Reply:
I know what it does. I'm asking why did you change it from human readable to human unreadable.

Reply:
This is in preparation for #2869, where `to_hex` will do a call-through to `to_string`. It makes sense to have something human-readable for `to_string`, but I dont think what's currently implemented will be what is expected when calling `to_hex`. You also mentioned [something similar about returning pure hex](https://github.com/rust-bitcoin/rust-bitcoin/issues/2869#issuecomment-2206381458) - not sure if this is what you meant as well. If it makes more sense to not have the call-through to `to_string`, that's also fine; we can update #2869 to reflect that

Reply:
> If it makes more sense to not have the call-through to `to_string`

Indeed, it does, I think. Similar to how `Script` displays as ASM and has a `to_hex` method.

Reply:
yep, though this type falls under the exception [you mentioned](https://github.com/rust-bitcoin/rust-bitcoin/pull/2318#discussion_r1453062815):

> Only for types with unknown encoded lengths (e.g. all consensus types containing a `Vec`)

I opened this issue to prep for that call through since `Key` has the `Vec` field of `key`. I'm not sure how often `to_string` is being used for `Key`, but if `to_hex` for other types are call throughs to `to_string`, I'd lean more in that direction here as well for consistency

Reply:
I can't remember why I said it. We can trivially pre-allocate here - `String::with_capacity(self.key_len * 2 + 2)`.

Reply:
Is this what's going to be returned for `Display`? Why do we need `self.key_len * 2 + 2`? is there a scenario where we would have more data than `<key> := <keylen> <keytype> <keydata>`?

Reply:
No, that's in `to_hex`. `self.key_len * 2 + 2` is literally the length of the resulting string.